### ISSUE TITLE
there was something wrong when the program's language in chinese

### ISSUE THREAD

**Issue by Cccc-owo** [Earthcomputer/EnchantmentCracker#145]
![image](https://user-images.githubusercontent.com/47687154/78141692-6f395e80-745e-11ea-98c4-7077fe3e8b8e.png)
![image](https://user-images.githubusercontent.com/47687154/78141696-71032200-745e-11ea-9518-bb5c92938401.png)
![image](https://user-images.githubusercontent.com/47687154/78141697-719bb880-745e-11ea-905f-e7709c2b8726.png)
In some places, it doesn't display text properly.Just like the pictures

**Comment by Earthcomputer** [comment]
It looks like your font, but I will try and mitigate the problem before the 1.16 release

**Comment by Stehsaer** [comment]
see if you can change the font to Microsoft YaHei UI
把字体换成微软雅黑兼容性可能会好一点

**Comment by Earthcomputer** [comment]
It's Java's responsibility to pick the correct font. I am currently asking for a sans serif font if the Minecraft font cannot display all the characters in a text. Perhaps I should ask for the font which Java normally uses for dialogs.

**Comment by Earthcomputer** [comment]
@Cccc-u can you try building from the latest commit and see if the issue persists?

**Comment by Earthcomputer** [comment]
Or @Stehsaer you could try as well?

**Comment by Cccc-u** [comment]
@Earthcomputer  it still has the problem

**Comment by Cccc-u** [comment]
@Earthcomputer
hi,i remade a language file.
i tested it and it works fine on my PC.is it OK?
[EnchantmentCracker_zh.zip](https://github.com/Earthcomputer/EnchantmentCracker/files/4469669/EnchantmentCracker_zh.zip)

**Comment by Stehsaer** [comment]
Is it possible to duplicate the whole project to C#(.net) that may help because it may be easier to deal with problems like this (creating a window that looks like a current one may be alittle harder... @Earthcomputer

**Comment by Earthcomputer** [comment]
@Stehsaer of course, you can do that any time (i'm not going to myself though).
If anyone wants to find a fix to the issue where the UTF-8 file actually loads properly on Windows (it seems to be a combination of the operating system and the font), I would be very grateful.

**Comment by Stehsaer** [comment]
@Earthcomputer To be honest I don't know java much and i have no way to figure out how the cracker work. Can you provide how this thing works exactly and maybe i can make it in C#. thanks

**Comment by Earthcomputer** [comment]
I explained how it works in the last 5 minutes of the video linked in the readme. You may need to read the enchantment cracker code or Minecraft code to get every detail. Also @Hexicube said something about making a .NET version, you should talk to him about it.

**Comment by Earthcomputer** [comment]
I think I have managed to fix this issue by converting to the escape sequence format dynamically as the translations are loaded. This means we should get the best of both worlds, easy files for translators to edit, and support on more Java versions and systems, at the cost of [just a few lines of slightly disgusting code](https://github.com/Earthcomputer/EnchantmentCracker/blob/master/src/enchcracker/UTF8ResourceBundleControl.java). I'll create a release tomorrow with the 1.16 release, let me know whether this problem still persists.

**Comment by Earthcomputer** [comment]
Does this issue still persist with the latest release, 1.5?
@Cccc-owo

**Comment by Cccc-owo** [comment]
@Earthcomputer  
It seems not good on my PC
![批注 2020-06-24 145352](https://user-images.githubusercontent.com/47687154/85510923-9f8e3480-b62a-11ea-90ac-b7a1253bf6c2.png)
![批注 2020-06-24 145410](https://user-images.githubusercontent.com/47687154/85510930-a1f08e80-b62a-11ea-84e3-17064857cbae.png)

**Comment by Earthcomputer** [comment]
This is really weird. It should now be dynamically loading the same language file which you claim to work on your PC.

**Comment by Earthcomputer** [comment]
@Cccc-owo but I don't want your language file, because raw unicode escape sequences are hard for translators to edit. I want a solution where my language file can be loaded. Can you tell me what is your Java version, operating system and maybe your log file too?

**Comment by Cccc-owo** [comment]
> @Cccc-owo but I don't want your language file, because raw unicode escape sequences are hard for translators to edit. I want a solution where my language file can be loaded. Can you tell me what is your Java version, operating system and maybe your log file too?

Well,i understand.
i'm using Java 14.0.1 and Windows 10 1909.
This is my log File in `\bin`

**Comment by Earthcomputer** [comment]
Sorry, what is your log file?

**Comment by Earthcomputer** [comment]
This just happened to someone with the Ł character in the Polish language

**Comment by Cccc-owo** [comment]
it's fine,thanks

**Comment by Earthcomputer** [comment]
No, just because you have fixed it for yourself doesn't mean that the problem has gone away.

**Comment by ImUrX** [comment]
Shouldn't you try using a noto font? maybe one of the CJK ones
